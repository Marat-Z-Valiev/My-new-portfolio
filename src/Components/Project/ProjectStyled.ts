import styled from "styled-components";

const ProjectStyled = styled.div`
	.white {
		clip-path: polygon(0 0%, 100% 4%, 100% 96%, 0% 100%);
		background-color: #ffffff;
	}

	.blue {
		clip-path: polygon(0 8%, 100% 4%, 100% 96%, 0% 100%);
		background-color: #2891de;
	}

	.project {
		padding-top: 50px;
		padding-bottom: 50px;

		.project-info {
			display: flex;
			justify-content: space-evenly;

			@media (max-width: 996px) {
				flex-direction: column;
			}

			.right {
				order: 2;
			}

			img {
				box-shadow: none;
				margin-left: 15px;
				width: 100%;
				height: auto;
				box-shadow: 10px 7px 52px -1px rgba(0, 0, 0, 0.37);

				@media (max-width: 996px) {
					order: 1;
					width: 90%;
					margin: 0 auto;
				}

				@media (min-width: 1366px) {
					width: 75%;
					width: 800px;
					margin: 0 auto;
				}
			}
		}
		.project-description {
			display: flex;
			flex-direction: column;
			justify-content: space-evenly;
			align-items: center;
			justify-content: end;

			h2 {
				font-size: 1.8rem;
			}

			.project-summary {
				padding: 15px;
				font-size: 1.3rem;
			}
		}
		.technologies {
			display: flex;
			justify-content: center;

			img {
				width: 80px;
				height: 80px;
				margin-right: 15px;
			}
		}

		.buttons {
			display: flex;
			justify-content: center;
			width: 100%;
			margin-top: 40px;

			a {
				width: 100%;
				max-width: 200px;
				padding: 15px;
				margin-right: 15px;
				color: #25242e;
				text-decoration: none;
				text-transform: uppercase;
				text-align: center;
				background-color: #549657;
				color: white;
				transition: background-color ease-in 0.3s;

				@media (min-width: 320px) and (max-width: 425px) {
					max-width: 100px;
				}

				&:hover {
					background-color: #3d7140;
				}
			}
		}
	}
`;

export default ProjectStyled;
