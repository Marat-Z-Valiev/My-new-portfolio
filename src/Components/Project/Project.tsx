import React from "react";
import ProjectStyled from "./ProjectStyled";

interface ProjectProps {
	project: any;
}

const Project: React.FC<ProjectProps> = ({project}) => {
	const {
		name,
		description,
		image,
		websiteLink,
		githubLink,
		backgroundClass,
		position,
	} = project;
	const backgroundColor = backgroundClass === "white" ? "white" : "blue";
	const imagePosition = position === "right" ? "right" : "";
	return (
		<ProjectStyled>
			<div className={`project ${backgroundColor}`}>
				<div className="project-info">
					<img className={imagePosition} src={image} alt="" />
					<div className="project-description">
						<h2>{name}</h2>
						<p className="project-summary">{description}</p>
					</div>
				</div>
				<h2>Technologies</h2>
				<div className="technologies">
					<img src="https://i.ibb.co/qYDMK54/React-js-logo.png" alt="" />
					<img src="https://i.ibb.co/QCN8KZ9/redux.png" alt="" />
					<img src="https://i.ibb.co/Ct9gHzc/typescript.png" alt="" />
				</div>
				<div className="buttons">
					<a href={websiteLink} target="blank">
						Live Preview
					</a>
					<a href={githubLink} target="blank">
						View Github
					</a>
				</div>
			</div>
		</ProjectStyled>
	);
};

export default Project;
