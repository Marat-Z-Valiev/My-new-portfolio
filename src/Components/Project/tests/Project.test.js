import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Project from "../Project";

configure({adapter: new Adapter()});

describe("Project", () => {
	it("should render correctly", () => {
		const props = {
			project: {technologies: ["test"]},
		};
		const wrapper = shallow(<Project {...props} />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
