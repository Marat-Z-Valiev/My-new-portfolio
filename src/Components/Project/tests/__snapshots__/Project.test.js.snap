// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should render correctly 1`] = `
<ProjectStyled>
  <div
    className="project blue"
  >
    <div
      className="project-info"
    >
      <img
        alt=""
        className=""
      />
      <div
        className="project-description"
      >
        <h2 />
        <p
          className="project-summary"
        />
      </div>
    </div>
    <h2>
      Technologies
    </h2>
    <div
      className="technologies"
    >
      <img
        alt=""
        key="test"
        src="test"
      />
    </div>
    <div
      className="buttons"
    >
      <a
        target="blank"
      >
        Live Preview
      </a>
      <a
        target="blank"
      >
        View Github
      </a>
    </div>
  </div>
</ProjectStyled>
`;
